### PR TITLE
Adding broad phase algorithm for use with the DBVT.

### PR DESCRIPTION
**Pull request by Rhuagh**

### CODE REVIEW

File: src/algorithm/broad_phase/dbvt.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+use std::fmt::Debug;
+
+use dbvt::{DiscreteVisitor, DynamicBoundingVolumeTree, TreeValue};
+use prelude::*;
+
+/// DBVT collision testing
+pub struct DbvtBroadPhase;
+
+impl DbvtBroadPhase {
+    /// Create a new DbvtBroadPhase, borrows the tree to check against
```
Comment (vitvakatu) [inline src/algorithm/broad_phase/dbvt.rs]: Is doesn't really borrows the tree. Is doc string correct?

Comment (Rhuagh) [reply src/algorithm/broad_phase/dbvt.rs]: Ah, no, the first version borrowed the tree. I'll fix later

File: src/algorithm/broad_phase/dbvt.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
+//! DBVT broad phase collision test
```
Comment (kvark) [inline src/algorithm/broad_phase/dbvt.rs]: would be great to expand on what DBVT means here

File: src/algorithm/broad_phase/dbvt.rs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        for &(shape_node_index, ref shape) in tree.values() {
+            // We get the node index from the values list in the tree, and immediately get the value
+            // index of said node, so unwrapping will be safe for all cases where the tree isn't
+            // corrupt. The tree being corrupt is a programming error, which should be a panic.
+            let shape_value_index = tree.value_index(shape_node_index).unwrap();
+            if dirty[shape_value_index] {
+                for (hit_value_index, _) in
+                    tree.query_for_indices(&mut DiscreteVisitor::<T::Bound, T>::new(shape.bound()))
+                {
+                    match match shape_value_index.cmp(&hit_value_index) {
```
Comment (kvark) [inline src/algorithm/broad_phase/dbvt.rs]: hahah, wow. Let's add a temporary variable here to avoid confusion?

**Comment by Rhuagh** [comment]
There we go.

**Comment by kvark** [comment]
Thank you!
bors r+

**Pull request merged**